fix: stg Redis 연결 실패로 OAuth 로그인 후 500 발생

### 파트
be

### 증상
stg 환경에서 `/oauth2/authorization/google` 로그인 플로우를 끝까지 진행하면, 콜백(`/login/oauth2/code/google`) 단계에서 `RefreshTokenRepository.save`가 Redis에 refresh token을 쓰려다 `RedisConnectionFailureException: Unable to connect to Redis (localhost/127.0.0.1:6379)` 가 발생. 결과적으로 사용자에게 Whitelabel error page 가 노출되고 로그인이 완료되지 않음.

원인은 `application-stg.properties`에 redis 호스트가 `localhost`로 하드코딩되어 있고, stg EC2/컨테이너에는 Redis 인스턴스 자체가 떠있지 않기 때문.

```
spring.data.redis.host=localhost
spring.data.redis.port=6379
```

### 재현 방법
1. stg FE에서 구글 로그인 진입 (`/oauth2/authorization/google`)
2. 구글 인증 통과 후 `https://api-stg.asklogue.co/login/oauth2/code/google?...` 콜백 도달
3. Whitelabel error page 노출 확인
4. stg BE 로그에서 `RedisConnectionFailureException: Unable to connect to Redis ... Connection refused: localhost/127.0.0.1:6379` 확인

### 기대 동작
- OAuth 콜백 처리 → refresh token Redis 저장 성공 → JWT redirect base로 정상 리다이렉트
- stg/prod 양쪽 모두 Redis 호스트 정보가 환경변수로 주입되어야 하며, 실제 Redis 인스턴스가 같은 네트워크에 존재해야 함

### 환경
- OS: Amazon Linux (stg EC2, SSM 배포)
- 브랜치/커밋: `dev` 최신
- Spring Boot 3.x / Java 21, lettuce 6.6.0
- 발생 환경: stg (prod 동일 코드 경로라 동일 위험)

### 로그/스크린샷
```
2026-05-23T09:45:55.659Z ERROR ... org.springframework.data.redis.RedisConnectionFailureException: Unable to connect to Redis
    at ... RefreshTokenRepository.save(RefreshTokenRepository.java:34)
    at ... OAuth2LoginSuccessHandler.onAuthenticationSuccess(OAuth2LoginSuccessHandler.java:125)
Caused by: io.lettuce.core.RedisConnectionException: Unable to connect to localhost/<unresolved>:6379
Caused by: java.net.ConnectException: Connection refused
```

### 작업 항목

- [ ] **BE 코드**: `application-stg.properties` (그리고 `application-prod.properties` 동일하게) Redis 설정 환경변수화
  ```properties
  spring.data.redis.host=${REDIS_HOST:localhost}
  spring.data.redis.port=${REDIS_PORT:6379}
  spring.data.redis.password=${REDIS_PASSWORD:}
  ```

- [ ] **배포 파이프라인** (`.github/workflows/be-deploy.yml`): SSM commands에 공유 docker network + redis 컨테이너 보장 단계 추가
  1. `docker network inspect logue-net || docker network create logue-net`
  2. redis 컨테이너 멱등 실행 (이미 떠있으면 건드리지 않음)
     ```bash
     docker run -d --name redis --restart unless-stopped \
       --network logue-net \
       -v redis-data:/data \
       redis:7-alpine redis-server --appendonly yes --requirepass "$REDIS_PASSWORD"
     ```
  3. BE 컨테이너에 `--network logue-net` 추가

- [ ] **SSM Parameter Store**: `/logue/stg/` (와 `/logue/prod/`) 에 키 추가
  - `REDIS_HOST=redis`
  - `REDIS_PORT=6379`
  - `REDIS_PASSWORD=<강한 비밀번호>`

### 검증 계획
- 배포 후 stg FE 로그인 재시도 → 200 redirect + JWT 발급 확인
- stg EC2에서 `docker exec -it redis redis-cli -a $REDIS_PASSWORD PING` → `PONG` 확인
- BE 컨테이너에서 redis 컨테이너로 ping 가능한지 확인
- prod도 같은 패턴 적용 후 동일 검증

### 리스크
- 첫 배포 시 BE 컨테이너 네트워크 옵션 변경으로 한번 짧은 다운타임 발생 가능
- redis 비밀번호가 SSM Parameter Store에만 존재해야 함 (코드/리포지토리 누설 금지)
- redis 데이터는 `redis-data` 볼륨에 보존되지만, EC2 자체가 교체되면 유실. stg는 손실 무방 (refresh token이라 재로그인이면 끝)

--- a/.github/workflows/be-deploy.yml
+++ b/.github/workflows/be-deploy.yml
@@ -89,9 +89,12 @@ jobs:
               "aws ecr get-login-password --region $AWS_REGION | docker login --username AWS --password-stdin $REGISTRY",
               "docker pull $IMAGE",
               "aws ssm get-parameters-by-path --region $AWS_REGION --path /logue/$ENV/ --recursive --with-decryption --output json --query 'Parameters[*]' | python3 -c \"import sys,json; params=json.load(sys.stdin); open('/tmp/.logue.env','w').write(''.join([p['Name'].split('/')[-1]+'='+p['Value']+chr(10) for p in params]))\"",
+              "set -a && . /tmp/.logue.env && set +a",
+              "docker network inspect logue-net >/dev/null 2>&1 || docker network create logue-net",
+              "docker ps --filter name=^redis\$ --format '{{.Names}}' | grep -q '^redis\$' || docker run -d --name redis --restart unless-stopped --network logue-net -v redis-data:/data redis:7-alpine redis-server --appendonly yes --requirepass \"\$REDIS_PASSWORD\"",
               "docker stop $CONTAINER_NAME 2>/dev/null || true",
               "docker rm   $CONTAINER_NAME 2>/dev/null || true",
-              "docker run -d --name $CONTAINER_NAME --restart unless-stopped -p $APP_PORT:8080 -e SPRING_PROFILES_ACTIVE=$ENV --env-file /tmp/.logue.env $IMAGE",
+              "docker run -d --name $CONTAINER_NAME --restart unless-stopped --network logue-net -p $APP_PORT:8080 -e SPRING_PROFILES_ACTIVE=$ENV --env-file /tmp/.logue.env $IMAGE",
               "rm -f /tmp/.logue.env",
               "docker image prune -f"
             ]

--- a/apps/be/src/main/resources/application-prod.properties
+++ b/apps/be/src/main/resources/application-prod.properties
@@ -69,5 +69,12 @@ logue.storage.s3.access-key=${LOGUE_STORAGE_S3_ACCESSKEY:}
 logue.storage.s3.secret-key=${LOGUE_STORAGE_S3_SECRETKEY:}
 
 # ================================
+# Redis
+# ================================
+spring.data.redis.host=${REDIS_HOST:localhost}
+spring.data.redis.port=${REDIS_PORT:6379}
+spring.data.redis.password=${REDIS_PASSWORD:}
+
+# ================================
 # Sentry
 # ================================

--- a/apps/be/src/main/resources/application-stg.properties
+++ b/apps/be/src/main/resources/application-stg.properties
@@ -58,6 +58,7 @@ logue.storage.s3.secret-key=${LOGUE_STORAGE_S3_SECRETKEY:}
 # ================================
 # Redis
 # ================================
-spring.data.redis.host=localhost
-spring.data.redis.port=6379
+spring.data.redis.host=${REDIS_HOST:localhost}
+spring.data.redis.port=${REDIS_PORT:6379}
+spring.data.redis.password=${REDIS_PASSWORD:}
 

